app/controllers/api/v1/users_controller.rb

```diff
@@ -10,6 +10,16 @@ def index
         render json: UserSerializer.new(paginate(users)).serializable_hash.to_json
       end
 
+      def create
+        user = User.find_or_create_by!(email: params[:data][:attributes][:email]) do |u|
+          u.full_name = params[:data][:attributes][:full_name]
+        end
+
+        hash = UserSerializer.new(user).serializable_hash
+
+        render json: hash, status: :created
+      end
```

Comment:
I don't know what your requirements are, but I find this surprising that we respond with 201 created when we match an existing record. I'm not sure what would be better here, perhaps just 200, or maybe 409?

Reply:
i can do the extra check and return 201 if it is new and 200 if it already exists

Reply:
Only if you think it would be useful to distinguish, just wanted to make sure it was considered 🙂

Reply:
- ok i changed my mind. so i kind of purposefully glossed over this particular implementation detail as I know it would open a can of worms.
- `409` works better i feel because the `full_name` field is also passed over
- I feel there is ambiguity if a 200 be returned whether `full_name` gets updated or not
- so instead we simply return a `409` if a user is found with the email address given. what do you think? 
- the current implementation lacks but i think is fine for now. it is missing error info and should have but will implement as a later date the ability to link to the conflicting record so the client can fetch that existing record instead. we do not have such an api endpoint so this also would not be possible.